go/tasks/plugins/array/catalog.go

```diff
@@ -389,7 +419,22 @@ func ConstructCatalogReaderWorkItems(ctx context.Context, taskReader core.TaskRe
 	return workItems, nil
 }
 
-func ConstructInputReaders(ctx context.Context, dataStore *storage.DataStore, inputPrefix storage.DataReference,
+// ConstructStaticInputReaders constructs input readers that comply with the io.InputReader interface but have their
+// inputs already populated.
+func ConstructStaticInputReaders(_ context.Context, inputPaths io.InputFilePaths, inputs *idlCore.LiteralCollection, inputName string) ([]io.InputReader, error) {
```

Comment:
nit: no need to have an error in the response signature since you never return one?